src/Microsoft.Kusto.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1594,69 +896,38 @@ internal SignatureHelp GetSignatureHelp(TextDocumentPosition textDocumentPositio
             if (scriptParseInfo == null)
             {
                 var scriptDocInfo = ScriptDocumentInfo.CreateDefaultDocumentInfo(textDocumentPosition, scriptFile);
-                resultCompletionItems = AutoCompleteHelper.GetDefaultCompletionItems(scriptDocInfo, useLowerCaseSuggestions);
-                //call completion extensions only for default completion list
-                resultCompletionItems = await ApplyCompletionExtensions(connInfo, resultCompletionItems, scriptDocInfo);
+                resultCompletionItems = resultCompletionItems = AutoCompleteHelper.GetDefaultAutoComplete(DataSourceType.Kusto, scriptDocInfo, textDocumentPosition.Position);
                 return resultCompletionItems;
             }
 
             ScriptDocumentInfo scriptDocumentInfo = new ScriptDocumentInfo(textDocumentPosition, scriptFile, scriptParseInfo);
 
-            // reparse and bind the SQL statement if needed
-            if (RequiresReparse(scriptParseInfo, scriptFile))
-            {
-                ParseAndBind(scriptFile, connInfo);
+            if(connInfo != null){
+                ReliableDataSourceConnection connection;
+                connInfo.TryGetConnection("Default", out connection);
+                IDataSource dataSource = connection.GetUnderlyingConnection();
+			    
+                resultCompletionItems = dataSource.GetAutoCompleteSuggestions(scriptDocumentInfo, textDocumentPosition.Position);
+            }
+            else{
+                resultCompletionItems = AutoCompleteHelper.GetDefaultAutoComplete(DataSourceType.Kusto, scriptDocumentInfo, textDocumentPosition.Position);
```

Comment:
Is there some way to get the data source generically here rather than hard code to Kusto?

Reply:
Added a comment here for tracking it later.